tests/controls.js

```diff
@@ -95,72 +75,73 @@
       .twoway();
 
     equal(e._vx, 0, "vx starts equal to 0");
-    equal(e._x, 0, "No change in _x when twoway is called");
 
     e.enableControl();
     e.speed({ x: 50, y: 50 });
     Crafty.timer.simulateFrames(1);
     equal(e._vx, 0, "No change in speed after Twoway speed is set");
-    equal(e._x, 0);
 
-    Crafty.trigger('KeyDown', {
-      key: Crafty.keys.D
-    });
+    e.disableControl();
+    keysDown(Crafty.keys.D);    
     Crafty.timer.simulateFrames(1);
-    equal(e._vx, 50, "vx = 50 when key D pressed");
-    equal(e._vy, 0, "vy = 0 when key D pressed");
+    equal(e._vx, 0, "vx = 0 when key D pressed while control is disabled");
+
+    e.enableControl();
+    Crafty.timer.simulateFrames(1);    
+    equal(e._vx, 50, "vx = 50 once control is enabled while D key is down");
 
     e.disableControl();
-    equal(e._vx, 0, "vx = 0 once control is disabled");
+    keysUp(Crafty.keys.D);
+    Crafty.timer.simulateFrames(1); 
+    equal(e._vx, 50, "vx = 50 when key is released while control disabled");
 
-    Crafty.trigger('KeyUp', {
-      key: Crafty.keys.D
-    });
-    equal(e._vx, 0, "vx = 0 when key is released while control is disabled");
+    e._vx = 17;
+    Crafty.timer.simulateFrames(1);
+    equal(e._vx, 17, "vx = 17 after being explicitly set while control disabled");
```

Comment:
This is the one point where the behavior has deliberately changed: disabling control no longer resets the velocity of the entity.  Easy enough to set velocity to 0 if that's what you want when calling disable control.